### ISSUE TITLE
ebay-textbox: helper text character count alignment

### ISSUE THREAD

**Issue by randybascue** [eBay/skin#2047]
## Problem
When there are multiple lines of helper text and a character counter, the character counter is vertically centered.

## Expected
The character counter is aligned to the baseline of the first line of helper text (top-aligned).

## Current example
![Screen Shot 2023-04-14 at 1 09 48 PM](https://user-images.githubusercontent.com/35155833/232141312-a7c71488-8669-4612-b3a8-cf901b78db02.png)

## Expected example
![image](https://user-images.githubusercontent.com/35155833/232144887-bd6bb5a0-64bf-4806-bc95-96e5e11fc34f.png)

## Specs
![image](https://user-images.githubusercontent.com/35155833/232145103-5501c5dd-5530-41c9-a984-79a7f785ab3c.png)

**Comment by agliga** [comment]
I don't believe we have this built, but we can probably change the post text to be top aligned.

**Comment by ArtBlue** [comment]
@randybascue , there are open questions about this. We'll need to punt it to `16.3.0`.

**Comment by ArtBlue** [comment]
Blocking this...the issue says "ebay-textbox," but that component doesn't have any text counter. Ebay-field might be the component in question, but that doesn't have the text (counter) on the right side either - so nothing to align the left to.

**Comment by randybascue** [comment]
I'm not seeing ebay-field on Storybook so I referenced ebay-textbox.

**Comment by randybascue** [comment]
Anything that has the character counter should be aligned like this. I can update the title to the name of whichever component has the character counter but I'm not seeing a textarea component. And setting the `invalid` boolean to true on ebay-textbox doesn't show helper text at all.

**Comment by ArtBlue** [comment]
@randybascue , that "error/'invalid" state seems to be working in ebay ui core (not seeing any issue). 
<img width="1406" alt="image" src="https://github.com/eBay/skin/assets/1675667/bd02485f-e796-4fbe-ba78-8b5d94b57ef1">

**Comment by randybascue** [comment]
@ArtBlue is there a reason it's not included in the specs? Do we need another ticket to add that functionality?

**Comment by agliga** [comment]
So to add a bit of context here.
We have a field module in skin here: https://opensource.ebay.com/skin/#field
This is what provides the end text count in textbox. There is a `field-description--end` which we use as a modifier to add text to the end of the textbox. We don't provide it as part of textbox automatically. We have an issue for that but never got around to it. Teams have been adding it manually using `field` 
We did have it in the docs but it seems to not be there.

We don't want it in ebayui at the moment because we don't want components that deal with layout in ebayui. 

I hope this clarifies it.

**Comment by ArtBlue** [comment]
@randybascue , based on the context @agliga provided, the reason why the text counter is not included in the Skin specs/docs is that it doesn't exist. I actually doubt whether it ever existed - I don't see any reference in git log, I don't see any old Skin docs snapshot with it (https://web.archive.org/web/20220703061920/https://opensource.ebay.com/skin/#field).

I just checked the Skin issues and I'm not finding any reference to "field" with the text counter to be added or any left or  right-aligned helper text to be added to the component. 

These screenshots you have are likely from teams creating their own thing. If so, the near-term fix they would be for them to apply a spacing fix on their end.

I'd say the way forward with this is to either repurpose this issue to add the feature(s) to the `field` component or close this issue and create a new one for the addition of the features. It would obviously include the proper alignment and the addition of the left-aligned helper text and the right-aligned text counter.

**Comment by ArtBlue** [comment]
I updated the issue to correctly reflect the feature additions.

**Comment by agliga** [comment]
@randybascue Can we get max wdiths for this? So we can wrap them properly. Thanks!

**Comment by ianmcburnie** [comment]
Yes, this will be an update/addition to the field module rather than the textbox on the skin side. Once skin side is done we can update coreui to leverage it.

**Comment by randybascue** [comment]
@agliga the width of the helper text would match the field with 16px separating character count and helper text if both are visible

**Comment by ArtBlue** [comment]
@ianmcburnie , since I'll be taking over this and you had some specific knowledge regarding approach, when you have a moment, can you please relay some of that info to provide direction? I have a mental model of approach, but I want to have it consistent with the approach you were going to take.

**Comment by ianmcburnie** [comment]
The field modules gives basic defaults for the most common label + form control layout variations using classes alone. I think the current design of field should accommodate this layout variation with only minimal changes/additions.

I would have a good read through the documentation and familiarize yourself with the main design approach of unstacked field (default) vs stacked field (modifier), inline label (default) vs stacked label (modifier), auto width (default) vs fluid width (modifier) and the various permutations of those:

https://opensource.ebay.com/skin/#field

I think what we can do is expand on the current `field__description` to allow left and right portions. Or if that doesn't work, just create a bespoke `field__character-meter` class purely for this purpose.
